moves/tests.py

```diff
@@ -1,3 +1,81 @@
+import pdb
+import json
+import datetime
+from django.utils import timezone
+from django.test import override_settings
 from django.test import TestCase
+from .models import Move, Spot
 
 # Create your tests here.
+class IndexTests(TestCase):
+    def test_uuid_is_set_after_navigating_to_the_index_page(self):
+        self.assertIsNone(self.client.session.get('user_uuid'))
+        self.client.get('/')
+        self.assertTrue(bool(self.client.session.get('user_uuid')))
+
+    def test_user_uuid_is_not_reset_upon_navigation_to_index_page_if_set(self):
+        self.client.get('/')
+        uuid = self.client.session.get('user_uuid')
+        self.assertTrue(bool(uuid))
+        self.client.get('/')
+        self.assertEqual(self.client.session.get('user_uuid'), uuid)
+
+    @override_settings(DEBUG=True)
+    def test_bootstrap_model_on_page_if_model_with_uuid_with_past_six_hours_exists(self):
+        self.client.get('/')
+        uuid = self.client.session.get('user_uuid')
+        self.client.post('/json/spots/', { 'name': 'The Spot' })
+        self.client.post('/json/moves/', {
+            'user':'user',
+            'spot': Spot.objects.first().id,
+            'time': timezone.now() - datetime.timedelta(hours=3)
+        })
+        recent_move = json.loads(self.client.get('/').context['recent_move'])
+        self.assertEqual(uuid, recent_move['uuid'])
+        self.assertEqual(len(recent_move), 5)
+
+    @override_settings(DEBUG=True)
+    def test_users_name_bootrapped_if_model_beyond_six_hours_ago_exists(self):
+        self.client.get('/')
+        self.client.post('/json/spots/', { 'name': 'The Spot' })
+        self.client.post('/json/moves/', {
+            'user':'user',
+            'spot': Spot.objects.first().id,
+            'time': timezone.now() - datetime.timedelta(hours=9)
+        })
+        recent_move = json.loads(self.client.get('/').context['recent_move'])
+        self.assertEqual(len(recent_move), 1)
```

Comment:
Should probably test that the one field that is present is in fact `user` and that `id` or `spot` are not present.